api/base/exceptions.py

```diff
@@ -36,11 +43,30 @@ def json_api_exception_handler(exc, context):
     return response
 
 
+class JSONAPIException(APIException):
+    """Inherits from the base DRF API exception and adds extra metadata to support JSONAPI error objects
+
+    :param str detail: a human-readable explanation specific to this occurrence of the problem
+    param dict source: an object containing references to the source of the error, optionally including
+    any of the following members:
+    :parm str dict.pointer: a JSON Pointer [RFC6901] to the associated entity in the request document
+    e.g. "/data" for a primary data object, or "/data/attributes/title" for a specific attribute
+    :param str dict.parameter: a string indicating which URI query parameter caused the error
+    """
+    def __init__(self, detail=None, source=None):
+        super(JSONAPIException, self).__init__(detail=detail)
+        self.source = source
+
 # Custom Exceptions the Django Rest Framework does not support
 class Gone(APIException):
     status_code = status.HTTP_410_GONE
     default_detail = ('The requested resource is no longer available.')
 
+class InvalidQueryStringValue(JSONAPIException):
+    """Raised when client passes an invalid value to a querystring parameter."""
+    default_detail = 'Querystring contains an invalid value.'
+    status_code = http.BAD_REQUEST
+
```

Comment:
As suggested in a previous comment, the constuctor should take `parameter` as an argument--not `source`.

I think it would make sense for `JSONAPIErrors` to define how they get formatted as an error object. 

```python
err = InvalidQueryStringError(parameter='title')
err.to_error_object()
# {
#     'detail': 'Query string contains an invalid value.',
#     'source': {'parameter': 'title'},
# }


# TODO: Our custom Validation error that subclasses JSONAPIError,
# raised upon deserialization
err = InvalidAttributeError(attribute='category')
err.to_error_object()
# {
#     'detail': '...',
#     'source': {'pointer': '/data/attributes/category'},
# }
```